### ISSUE TITLE
On CentOS 7 : Should packetfence activate persistent journald logging at installation? (not activated by default)

### ISSUE THREAD

**Issue by tlaurion** [inverse-inc/packetfence#1571]
Default in journald.conf: 

> [Journal]
> Storage=auto

"auto" meaning here:

> One of "volatile", "persistent", "auto" and "none". If "volatile", journal log data will be stored only in memory, i.e. below the /run/log/journal hierarchy (which is created if needed). If "persistent", data will be stored preferably on disk, i.e. below the /var/log/journal hierarchy (which is created if needed), with a fallback to /run/log/journal (which is created if needed), during early boot and if the disk is not writable. "auto" is similar to "persistent" but the directory /var/log/journal is not created if needed, so that its existence controls where log data goes.

Solution to be implemented for el7 in the spec file in post-install:

```
mkdir -p /var/log/journal
systemd-tmpfiles --create --prefix /var/log/journal
systemctl restart systemd-journald
```

Justifications:

> Apparently, the default was changed in Fedora 19 (to persitent storage) and since CentOS 7 is derived from Fedora 18 - it is still non-persisent there, by default. Persistency is implemented by default outside of journald via /var/log/messages and the rotated versions /var/log/messages-YYYYMMDD which are written by rsyslogd (which runs by default and gets its input from journald).[Source](https://unix.stackexchange.com/questions/159221/how-display-log-messages-from-previous-boots-under-centos-7)

**Comment by julsemaan** [comment]
Status of this ?

Should we discuss in this issue, in a meeting or postpone to 6.4 ?

Doesn't seem critical

**Comment by tlaurion** [comment]
Becomes critical to me. Trying to debug #2140 requires this. Any debugging will require this, post hoc.
@louismunro requires it to have a size limit.

**Comment by julsemaan** [comment]
For sure we'll want a size limit on this, especially now that all the logging of PacketFence will go through journald via systemd

**Comment by louismunro** [comment]
The default size limit is 4Gib which seems reasonable to me.

**Comment by louismunro** [comment]
fixed in d6ca92eef1788a4e6d2bbb9a349659ae8b40c5c8
